akka-actor/src/main/java/akka/actor/dungeon/AbstractActorCell.java

```diff
@@ -17,20 +17,24 @@
     try {
       mailboxOffset =
           Unsafe.instance.objectFieldOffset(
-              ActorCell.class.getDeclaredField(
-                  "akka$actor$dungeon$Dispatch$$_mailboxDoNotCallMeDirectly"));
+              // TODO DOTTY
+              ActorCell.class.getDeclaredField("_mailboxDoNotCallMeDirectly"));
```

Comment:
Here I need some guidance on how to proceed, this AFAIU prevents today to use Akka Actor in a Scala 3 project.
But, at the same time, rely on the internals of the compiler, should we implement fallbacks or tackle the problem differently? cc. @raboof @johanandren @patriknw

Reply:
@eloots did you not hit this in https://github.com/eloots/akka-typed-on-scala-3 ?

Reply:
I assume that @eloots was using exclusively Akka Typed, probably not hitting the untyped ActorCell. But this needs to be proved ...

Reply:
The offsets here is used by the underlying impl also for typed actors (those are on top of "classic" actors), so that should be involved regardless. But Erics sample just used the pre-compiled 2.13 Akka artifacts rather than compiling them with Scala 3 himself, so maybe that keeps the encoding.

I guess this would have to be separate logic for Scala 2 and 3 somehow?

Reply:
I'll double-check that this is needed after the rest of the more "trivial" changes get merged, in case I'll go for the switch case. (Do we already have available the `scalaVersion` in the runtime, such as `BuildInfo` or something like `Version`?)

Reply:
Neither Scala 2 nor Scala 3 give any guarantee about the naming scheme of private fields, so while this might work on a particular version of a compiler, it might not work on the next.

Reply:
While I understand there is no such guarantee/promise, it has been stable in Scala 2 at least since 2012 (and Scala 2.10.0-M7). This will likely be tricky to solve if the internal encoding in Scala 3 is expected to change often. 

Perhaps moving on to VarHandle from Unsafe could solve it, but we can't do that as long as we also support Java 8 (I have a feeling that there was performance concerns around that as well).